tests/CppUTestExt/MockSupportTest.cpp

```diff
@@ -697,6 +717,21 @@ TEST(MockSupportTest, outputParameterSucceeds)
     CHECK_NO_MOCK_FAILURE();
 }
 
+TEST(MockSupportTest, customObjectOutputParameterSucceeds)
```

Comment:
There are two possible error cases that should be tested:
1. Comparator of wrong type. This does produce an error, but it should really get its own failure class.
2. No comparator found. This should produce an error, but can't, because we need to do memcpy in this case (might have been called via withOutputParameterReturning().
Because the "expected" results wouldn't be right in either case, I didn't implement either test case yet. Might use IGNORE_TEST() though, for now.